README.md

# Money Display

A Flutter widget for displaying Chinese currency format with flexible customization.

## Author

- **Author**: [Juncai Li](https://blog.csdn.net/qq_28550263?spm=1011.2415.3001.5343)
- **Email**: [291148484@163.com](291148484@163.com)
- **License**: [MIT](http://thispage.tech:9680/jclee1995/flutter_money_display/-/blob/master/LICENSE)
- Repository: [http://thispage.tech:9680/jclee1995/flutter_money_display.git](http://thispage.tech:9680/jclee1995/flutter_money_display.git)
- 中文文档：[http://thispage.tech:9680/jclee1995/flutter_money_display/-/blob/master/README_CN.md](http://thispage.tech:9680/jclee1995/flutter_money_display/-/blob/master/README_CN.md)

## Workflow Logic

This module provides a `ChineseMoneyDisplayWidget` component designed to display currency amounts in Chinese currency format, such as "¥123,456.78," and offers various customization options to meet different requirements.

The approximate logic for displaying currency in this module is as follows:

1. **Constructor Parameters**: The `ChineseMoneyDisplayWidget` constructor takes multiple parameters, including the currency amount (`price`) and various options for custom display, such as integer part font size, decimal part font size, maximum number, overflow symbol, and more.

2. **Formatting the Price**: The `_formatPrice` method is responsible for formatting the incoming `price` into rich text (`TextSpan`). First, it converts the price into a string and uses regular expressions to separate the integer and decimal parts.

3. **Handling Large Values**: If the price is greater than or equal to `maxNum`, it displays the `overflowSymbol`, typically a symbol representing infinity. If the price is less than `maxNum`, the processing continues.

4. **Handling Prices Greater Than Ten Thousand**: If the price is greater than ten thousand, the integer part is divided into an integer part and a decimal part (if any). If there is no decimal part, the "万" symbol is directly added. If there is a decimal part, both the integer and decimal parts are displayed together, with the integer part followed by the "万" symbol.

5. **Handling Prices Greater Than a Thousand**: If the price is greater than a thousand, the integer part is formatted, and a comma separator is added between thousands. The processing is similar to the above, depending on whether there is a decimal part, it displays the integer part, decimal part, and the "万" symbol.

6. **Handling Prices Less Than a Thousand**: For prices less than a thousand, the integer part does not have a comma separator. Instead, it displays the integer part, decimal part (if any), and the "万" symbol based on whether there is a decimal part.

7. **Handling Decimal Places**: If the price has a decimal part, it is displayed, and the style is set according to `smallFontsize` and `decimalDigitColor`.

8. **Text Integration**: All these text segments are integrated into a `TextSpan` for displaying together.

9. **Building Rich Text**: In the `build` method, a part of the rich text is the `currencySymbol`, which appears in front of the currency symbol. Then, the rich text part returned by `_formatPrice` is displayed in the `RichText` component. The entire component is responsible for displaying this rich text on the screen.

Through this logic, the `ChineseMoneyDisplayWidget` component can display the input price in Chinese currency format and customize the style based on the provided parameters. This allows developers to easily integrate it into Flutter applications to meet various currency display requirements.

## Usage

Describe the simplest usage in one sentence:

```dart
ChineseMoneyDisplayWidget(99.89)
```

To use the `MoneyDisplay` widget, simply import it and include it in your Flutter app. Here's a simple example:

```dart
import 'package:flutter/material.dart';
import 'package:money_display/money_display.dart';

void main() {
  runApp(const MoneyDisplayExample());
}

class MoneyDisplayExample extends StatelessWidget {
  const MoneyDisplayExample({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title:
              const Text('Money Display Example'), // Setting the app bar title.
        ),
        body: CustomScrollView(
          slivers: [
            const SliverToBoxAdapter(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  // Widget for displaying money with a euro symbol.
                  MoneyDisplayWidget(
                    192837465.2523432, // Set the price to be displayed.
                    currencySymbol:
                        CurrencySymbol.euro, // Set the currency symbol to euro.
                    showTrailingZeros:
                        false, // Determine whether to show trailing zeros.
                    currencySymbolSize:
                        26, // Set the size of the currency symbol.
                    currencySymbolColor: Color.fromARGB(255, 0, 43,
                        32), // Set the color of the currency symbol.
                    currencySymbolFontWeight: FontWeight
                        .w800, // Set the font weight of the currency symbol.
                    integerPartColor: Color.fromARGB(
                        255, 206, 192, 0), // Set the color of the integer part.
                    integerPartSize: 46, // Set the size of the integer part.
                    decimalPartColor:
                        Colors.purple, // Set the color of the decimal part.
                  ),
                  // Widget for displaying Chinese money with custom options.
                  ChineseMoneyDisplayWidget(
                    9999879999.92347,
                    maxNum: 10000000000, // Set a maximum number for display.
                    integerColor:
                        Colors.blue, // Set the color of the integer part.
                    integerFontsize:
                        25, // Set the font size of the integer part.
                    decimalFontsize:
                        21, // Set the font size of the decimal part.
                    currencySymbolColor:
                        Colors.pink, // Set the color of the currency symbol.
                    tenThousandSymbol: 'w', // Set the ten thousand symbol.
                    tenThousandSymbolColor: Color.fromARGB(255, 0, 156,
                        78), // Set the color of the ten thousand symbol.
                  ),
                  // Widget for displaying money with a custom currency symbol.
                  MoneyDisplayWidget(
                    12345, // Set the price to be displayed.
                    currencySymbol: CurrencySymbol
                        .custom, // Set the currency symbol to custom.
                    customCurrencySymbol:
                        '💲', // If using a custom currency symbol, set it here.
                    showTrailingZeros:
                        false, // Determine whether to show trailing zeros.
                  ),
                ],
              ),
            ),
            SliverGrid.count(
              crossAxisCount: 4,
              children: const [
                Center(child: ChineseMoneyDisplayWidget(0.9)),
                Center(child: ChineseMoneyDisplayWidget(0.17)),
                Center(child: ChineseMoneyDisplayWidget(0.1796)),
                Center(child: ChineseMoneyDisplayWidget(6)),
                Center(child: ChineseMoneyDisplayWidget(6.1)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    9.16,
                    decimalFontsize: 29,
                    integerColor: Color.fromARGB(255, 0, 150, 187),
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(9.1671)),
                Center(child: ChineseMoneyDisplayWidget(10)),
                Center(child: ChineseMoneyDisplayWidget(10.7)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    10.71,
                    integerFontsize: 25,
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(999)),
                Center(child: ChineseMoneyDisplayWidget(999.7)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    999.99,
                    decimalDigitColor: Color.fromARGB(255, 80, 0, 133),
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(1000)),
                Center(child: ChineseMoneyDisplayWidget(1995.0)),
                Center(child: ChineseMoneyDisplayWidget(1995.07)),
                Center(child: ChineseMoneyDisplayWidget(6666.66)),
                Center(child: ChineseMoneyDisplayWidget(9999)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    99999,
                    currencyFontWeight: FontWeight.w900,
                    currencySymbolColor: Color.fromARGB(255, 22, 0, 148),
                    currencySymbolFontsize: 56,
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(999999)),
                Center(child: ChineseMoneyDisplayWidget(996786)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                      9999999), // maxNum defaults to 10000.
                ),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    9999999,
                    maxNum:
                        10000001, // Set a custom maximum number for display.
                  ),
                ),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    9999999.97,
                    maxNum: 10000001,
                    integerColor: Colors.blue,
                    integerFontsize: 25,
                    decimalFontsize: 21,
                    currencySymbolColor: Colors.pink,
                    tenThousandSymbol: 'w',
                    tenThousandSymbolColor: Color.fromARGB(255, 0, 156, 78),
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}

```

![https://raw.githubusercontent.com/jacklee1995/flutter_money_display/master/example_OZYpI1Nuky.png](https://raw.githubusercontent.com/jacklee1995/flutter_money_display/master/example_OZYpI1Nuky.png)

## class ChineseMoneyDisplayWidget

In this example, we import the `ChineseMoneyDisplayWidget` and use it to customize the style and symbol for displaying prices.

### price

- **Type**: `double`
- **Default Value**: None
- **Description**: The price, the currency amount to be displayed in the widget.

### integerFontsize

- **Type**: `double`
- **Default Value**: `18`
- **Description**: Font size for the integer part.

### decimalFontsize

- **Type**: `double`
- **Default Value**: `15`
- **Description**: Font size for the decimal part.

### currencySymbolFontsize

- **Type**: `double`
- **Default Value**: `15`
- **Description**: Font size for the currency symbol.

### tenThousandSymbolSize

- **Type**: `double`
- **Default Value**: `15`
- **Description**: Font size for the ten-thousand separator ("万").

### maxNum

- **Type**: `double`
- **Default Value**: `1000000`
- **Description**: The maximum allowed number; if the number exceeds this value, the `overflowSymbol` will be displayed.

### overflowSymbol

- **Type**: `String`
- **Default Value**: `'-'`
- **Description**: The symbol to be displayed when the number exceeds the maximum value.

### tenThousandSymbol

- **Type**: `String`
- **Default Value**: `'万'`
- **Description**: The text used to separate ten-thousand units.

### currencySymbol

- **Type**: `String`
- **Default Value**: `'¥'`
- **Description**: The currency symbol, defaulting to the Chinese Renminbi symbol "¥."

### tenThousandSymbolColor

- **Type**: `Color`
- **Default Value**: Red
- **Description**: The color of the ten-thousand separator.

### integerColor

- **Type**: `Color`
- **Default Value**: Red
- **Description**: The text color for the integer part.

### decimalDigitColor

- **Type**: `Color`
- **Default Value**: Red
- **Description**: The text color for the decimal part.

### currencySymbolColor

- **Type**: `Color`
- **Default Value**: Red
- **Description**: The color of the currency symbol.

### overflowSymbolColor

- **Type**: `Color`
- **Default Value**: Red
- **Description**: The symbol color when the number exceeds the maximum value.

### integerFontWeight

- **Type**: `FontWeight`
- **Default Value**: `FontWeight.normal`
- **Description**: The font weight for the integer part.

### decimalFontWeight

- **Type**: `FontWeight`
- **Default Value**: `FontWeight.normal`
- **Description**: The font weight for the decimal part.

### tenThousandSymbolWeight

- **Type**: `FontWeight`
- **Default Value**: `FontWeight.normal`
- **Description**: The font weight for the ten-thousand separator ("万").

### currencyFontWeight

- **Type**: `FontWeight`
- **Default Value**: `FontWeight.normal`
- **Description**: The font weight for the currency symbol.

These parameters allow you to customize the appearance and behavior of the ChineseMoneyDisplayWidget to suit the needs of your application. You can control font size, color, symbols, and define the symbol to be displayed when the number exceeds the maximum value by adjusting these parameters.

## Class MoneyDisplayWidget

The `MoneyDisplayWidget` class is a component used for displaying formatted currency values in a customizable manner. It supports the following properties and methods:

### Properties

- `price` (double): The price to be displayed.
- `currencySymbol` (CurrencySymbol): The type of currency symbol, defaulting to [CurrencySymbol.yuan].
- `customCurrencySymbol` (String): If `currencySymbol` is set to `CurrencySymbol.custom`, this is the custom currency symbol used, defaulting to an empty string.
- `currencySymbolSize` (double): The size of the currency symbol, defaulting to 20.
- `currencySymbolColor` (Color): The color of the currency symbol, defaulting to red.
- `currencySymbolFontWeight` (FontWeight): The font weight of the currency symbol, defaulting to normal.
- `integerPartSize` (double): The size of the integer part, defaulting to 25.
- `integerPartColor` (Color): The color of the integer part, defaulting to red.
- `integerPartFontWeight` (FontWeight): The font weight of the integer part, defaulting to normal.
- `decimalPointSize` (double): The size of the decimal point, defaulting to 20.
- `decimalPointColor` (Color): The color of the decimal point, defaulting to red.
- `decimalPointFontWeight` (FontWeight): The font weight of the decimal point, defaulting to normal.
- `decimalPartSize` (double): The size of the decimal part, defaulting to 20.
- `decimalPartColor` (Color): The color of the decimal part, defaulting to red.
- `decimalPartFontWeight` (FontWeight): The font weight of the decimal part, defaulting to normal.
- `showTrailingZeros` (bool): Whether to display trailing zeros in the decimal part, defaulting to `false`.
- `customCurrencySymbol` (String): Allows for the use of a custom currency symbol, defaulting to an empty string.

### Methods

- `MoneyDisplayWidget({ ... })`: Constructor used for creating an instance of `MoneyDisplayWidget`.
- `build(BuildContext context)`: Build method, returns a `RichText` component used to display the formatted currency value.
- `formatPrice(double price, bool showTrailingZeros)`: Method for formatting prices, converting the price into a Map object containing the integer and decimal parts.

## Class CurrencySymbol

`CurrencySymbol` is an enumeration type used to define the type of currency symbol. It includes the following options:

- `CurrencySymbol.yuan`: Chinese Yuan symbol.
- `CurrencySymbol.dollar`: US Dollar symbol.
- `CurrencySymbol.euro`: Euro symbol.
- `CurrencySymbol.pound`: British Pound symbol.
- `CurrencySymbol.custom`: Custom currency symbol.
-


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## [1.0.1] - 2023-10-14

The default constructor parameter form of MoneyDisplayWidget is changed to:

```dart
 const MoneyDisplayWidget(
    this.price, {
    super.key,
    this.currencySymbol = CurrencySymbol.yuan,
    this.currencySymbolSize = 20,
    this.currencySymbolColor = Colors.red,
    this.currencySymbolFontWeight = FontWeight.normal,
    this.integerPartSize = 25,
    this.integerPartColor = Colors.red,
    this.integerPartFontWeight = FontWeight.normal,
    this.decimalPointSize = 20,
    this.decimalPointColor = Colors.red,
    this.decimalPointFontWeight = FontWeight.normal,
    this.decimalPartSize = 20,
    this.decimalPartColor = Colors.red,
    this.decimalPartFontWeight = FontWeight.normal,
    this.showTrailingZeros = false,
    this.customCurrencySymbol = '',
  });
```

## [1.0.0+2] - 2023-10-14

### Added

- Introducing the `MoneyDisplayWidget` class for displaying formatted currency values.
  - `price` (double): The price to be displayed.
  - `currencySymbol` (CurrencySymbol): The type of currency symbol.
  - `customCurrencySymbol` (String): Custom currency symbol if `currencySymbol` is set to `CurrencySymbol.custom`.

## [1.0.0+1] - 2023-10-14

### Added

- Introducing the `ChineseMoneyDisplayWidget` class for displaying Chinese-style formatted currency values.
  - `price` (double): The currency amount to be displayed.
  - `integerColor` (Color): The text color of the integer part.
  - `integerFontsize` (double): The font size of the integer part.
  - `decimalFontsize` (double): The font size of the decimal part.
  - `currencySymbolFontsize` (double): The font size of the currency symbol.
  - `tenThousandSymbolSize` (double): The size of the ten thousand symbol.
  - `decimalDigitColor` (Color): The text color of the decimal digits.
  - `maxNum` (double): The maximum allowed number; if exceeded, it displays the `overflowSymbol`.
  - `currencySymbol` (String): The currency symbol, typically "¥".
  - `overflowSymbol` (String): The symbol displayed when exceeding the maximum number (default: "-").
  - `tenThousandSymbol` (String): The text used to separate ten thousand units (default: "万").
  - `tenThousandSymbolColor` (Color): The color of the ten thousand unit symbol.
  - `currencySymbolColor` (Color): The color of the currency symbol.
  - `overflowSymbolColor` (Color): The color of the overflow symbol.
  - `integerFontWeight` (FontWeight): The font weight of the integer part.
  - `decimalFontWeight` (FontWeight): The font weight of the decimal part.
  - `tenThousandSymbolWeight` (FontWeight): The font weight of the ten thousand symbol.
  - `currencyFontWeight` (FontWeight): The font weight of the currency symbol.


lib/src/utils.dart
import 'package:flutter/material.dart';
import 'enums.dart';

/// 格式化整数部分，添加千分位分隔符
String formatWithComma(String price) {
  final parts = price.split('.');
  final wholePart = parts[0];
  final decimalPart = parts.length > 1 ? '.${parts[1]}' : '';
  final regex = RegExp(r'\B(?=(\d{3})+(?!\d))');
  return wholePart.replaceAllMapped(regex, (Match match) => ',') + decimalPart;
}

/// 创建富文本
TextSpan fText(
  String text,
  double fontSize,
  Color color,
  FontWeight fontWeight,
) {
  return TextSpan(
    text: text,
    style: TextStyle(
      fontSize: fontSize,
      color: color,
      fontWeight: fontWeight,
    ),
  );
}

String getCurrencySymbol(
    CurrencySymbol currencySymbol, String customCurrencySymbol) {
  if (currencySymbol == CurrencySymbol.custom) {
    return customCurrencySymbol;
  } else {
    return getCurrencySymbolFromEnum(currencySymbol);
  }
}

/// 枚举值转换为货币符号
String getCurrencySymbolFromEnum(CurrencySymbol symbol) {
  switch (symbol) {
    case CurrencySymbol.dollar:
      return '\$';
    case CurrencySymbol.euro:
      return '€';
    case CurrencySymbol.yuan:
      return '¥';
    case CurrencySymbol.franc:
      return '₣';
    case CurrencySymbol.ruble:
      return '₽';
    default:
      return '¥';
  }
}


lib/src/money_detail_widget.dart
/// ChineseMoneyDisplayWidget 是一个用于在Flutter应用程序中显示中国货币格式价格的组件
///
/// 作者：李俊才
/// 邮箱：291148484@163.com
/// 协议：MIT
///
/// 有关MIT协议的详细信息，请参阅项目的许可证文档（通常在项目的根目录中的"LICENSE"文件）
///
import 'package:flutter/material.dart';
import 'enums.dart';
import 'utils.dart';

/// 用于显示带有可自定义选项的格式化货币值的组件
class MoneyDisplayWidget extends StatelessWidget {
  /// 要显示的价格
  final double price;

  /// 货币符号的类型
  final CurrencySymbol currencySymbol;

  /// 如果[currencySymbol]设置为[CurrencySymbol.custom]，则使用自定义货币符号
  final String customCurrencySymbol;

  /// 货币符号的大小
  final double currencySymbolSize;

  /// 货币符号的颜色
  final Color currencySymbolColor;

  /// 货币符号的字体粗细
  final FontWeight currencySymbolFontWeight;

  /// 整数部分的大小
  final double integerPartSize;

  /// 整数部分的颜色
  final Color integerPartColor;

  /// 整数部分的字体粗细
  final FontWeight integerPartFontWeight;

  /// 小数点的大小
  final double decimalPointSize;

  /// 小数点的颜色
  final Color decimalPointColor;

  /// 小数点的字体粗细
  final FontWeight decimalPointFontWeight;

  /// 小数部分的大小
  final double decimalPartSize;

  /// 小数部分的颜色
  final Color decimalPartColor;

  /// 小数部分的字体粗细
  final FontWeight decimalPartFontWeight;

  /// 确定是否在小数部分显示尾随零
  final bool showTrailingZeros;

  /// 自定义构造函数用于 MoneyDisplayWidget
  ///
  /// [key] 是此组件的唯一键
  ///
  /// [price] 是要显示的价格值
  ///
  /// [currencySymbol] 定义货币符号的类型（默认为[CurrencySymbol.yuan]）
  ///
  /// [currencySymbolSize] 设置货币符号的大小（默认为20）
  ///
  /// [currencySymbolColor] 定义货币符号的颜色（默认为红色）
  ///
  /// [currencySymbolFontWeight] 设置货币符号的字体粗细（默认为FontWeight.normal）
  ///
  /// [integerPartSize] 确定整数部分的大小（默认为25）
  ///
  /// [integerPartColor] 定义整数部分的颜色（默认为红色）
  ///
  /// [integerPartFontWeight] 设置整数部分的字体粗细（默认为FontWeight.normal）
  ///
  /// [decimalPointSize] 设置小数点的大小（默认为20）
  ///
  /// [decimalPointColor] 定义小数点的颜色（默认为红色）
  ///
  /// [decimalPointFontWeight] 确定小数点的字体粗细（默认为FontWeight.normal）
  ///
  /// [decimalPartSize] 确定小数部分的大小（默认为20）
  ///
  /// [decimalPartColor] 设置小数部分的颜色（默认为红色）
  ///
  /// [decimalPartFontWeight] 设置小数部分的字体粗细（默认为FontWeight.normal）
  ///
  /// [showTrailingZeros] 指定是否在小数部分显示尾随零（默认为false）
  ///
  /// [customCurrencySymbol] 允许使用自定义货币符号（默认为空字符串）
  const MoneyDisplayWidget(
    this.price, {
    super.key,
    this.currencySymbol = CurrencySymbol.yuan,
    this.currencySymbolSize = 20,
    this.currencySymbolColor = Colors.red,
    this.currencySymbolFontWeight = FontWeight.normal,
    this.integerPartSize = 25,
    this.integerPartColor = Colors.red,
    this.integerPartFontWeight = FontWeight.normal,
    this.decimalPointSize = 20,
    this.decimalPointColor = Colors.red,
    this.decimalPointFontWeight = FontWeight.normal,
    this.decimalPartSize = 20,
    this.decimalPartColor = Colors.red,
    this.decimalPartFontWeight = FontWeight.normal,
    this.showTrailingZeros = false,
    this.customCurrencySymbol = '',
  });

  @override
  Widget build(BuildContext context) {
    // 格式化价格并提取整数和小数部分
    final formattedPrice = formatPrice(price, showTrailingZeros);
    final integerPart = formattedPrice['integerPart'];
    final decimalPart = formattedPrice['decimalPart'];

    // 创建要显示格式化价格的内联跨度列表
    List<InlineSpan> children = [
      TextSpan(
        text: getCurrencySymbol(currencySymbol, customCurrencySymbol),
        style: TextStyle(
          fontSize: currencySymbolSize,
          color: currencySymbolColor,
          fontWeight: currencySymbolFontWeight,
        ),
      ),
      TextSpan(
        text: formatWithComma(integerPart!),
        style: TextStyle(
          fontSize: integerPartSize,
          color: integerPartColor,
          fontWeight: integerPartFontWeight,
        ),
      ),
    ];

    // 如果适用，添加小数点和小数部分
    if (decimalPart!.isNotEmpty && decimalPart.length >= 2) {
      children.add(
        TextSpan(
          text: '.',
          style: TextStyle(
            fontSize: decimalPointSize,
            color: decimalPointColor,
            fontWeight: decimalPointFontWeight,
          ),
        ),
      );
      children.add(
        TextSpan(
          text: decimalPart.substring(0, 2),
          style: TextStyle(
            fontSize: decimalPartSize,
            color: decimalPartColor,
            fontWeight: decimalPartFontWeight,
          ),
        ),
      );
    }

    return RichText(
      text: TextSpan(
        children: children,
      ),
    );
  }

  /// 通过删除小数部分中的尾随零来格式化价格
  Map<String, String> formatPrice(double price, bool showTrailingZeros) {
    // 将价格转换为字符串
    final numberString = price.toString();
    // 定义用于匹配整数和小数部分的正则表达式
    final regex = RegExp(r'(\d+)(?:\.(\d+))?');
    // 尝试匹配正则表达式与数字字符串
    final match = regex.firstMatch(numberString);

    if (match != null) {
      final integerPartStr = match.group(1) ?? '';
      var decimalPartStr = match.group(2) ?? '';

      // 如果需要，从小数部分中删除尾随零
      if (!showTrailingZeros) {
        decimalPartStr = decimalPartStr.replaceAll(RegExp(r'0*$'), '');
      } else if (decimalPartStr.length == 1) {
        decimalPartStr += '0'; // 补充到有两位小数
      }

      // 返回格式化后的整数和小数部分
      return {'integerPart': integerPartStr, 'decimalPart': decimalPartStr};
    }

    // 如果没有找到匹配项，则返回空字符串
    return {'integerPart': '', 'decimalPart': ''};
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:money_display/money_display.dart';

void main() {
  runApp(const MoneyDisplayExample());
}

class MoneyDisplayExample extends StatelessWidget {
  const MoneyDisplayExample({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title:
              const Text('Money Display Example'), // Setting the app bar title.
        ),
        body: CustomScrollView(
          slivers: [
            const SliverToBoxAdapter(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  // Widget for displaying money with a euro symbol.
                  MoneyDisplayWidget(
                    192837465.2523432, // Set the price to be displayed.
                    currencySymbol:
                        CurrencySymbol.euro, // Set the currency symbol to euro.
                    showTrailingZeros:
                        false, // Determine whether to show trailing zeros.
                    currencySymbolSize:
                        26, // Set the size of the currency symbol.
                    currencySymbolColor: Color.fromARGB(255, 0, 43,
                        32), // Set the color of the currency symbol.
                    currencySymbolFontWeight: FontWeight
                        .w800, // Set the font weight of the currency symbol.
                    integerPartColor: Color.fromARGB(
                        255, 206, 192, 0), // Set the color of the integer part.
                    integerPartSize: 46, // Set the size of the integer part.
                    decimalPartColor:
                        Colors.purple, // Set the color of the decimal part.
                  ),
                  // Widget for displaying Chinese money with custom options.
                  ChineseMoneyDisplayWidget(
                    9999879999.92347,
                    maxNum: 10000000000, // Set a maximum number for display.
                    integerColor:
                        Colors.blue, // Set the color of the integer part.
                    integerFontsize:
                        25, // Set the font size of the integer part.
                    decimalFontsize:
                        21, // Set the font size of the decimal part.
                    currencySymbolColor:
                        Colors.pink, // Set the color of the currency symbol.
                    tenThousandSymbol: 'w', // Set the ten thousand symbol.
                    tenThousandSymbolColor: Color.fromARGB(255, 0, 156,
                        78), // Set the color of the ten thousand symbol.
                  ),
                  // Widget for displaying money with a custom currency symbol.
                  MoneyDisplayWidget(
                    12345, // Set the price to be displayed.
                    currencySymbol: CurrencySymbol
                        .custom, // Set the currency symbol to custom.
                    customCurrencySymbol:
                        '💲', // If using a custom currency symbol, set it here.
                    showTrailingZeros:
                        false, // Determine whether to show trailing zeros.
                  ),
                ],
              ),
            ),
            SliverGrid.count(
              crossAxisCount: 4,
              children: const [
                Center(child: ChineseMoneyDisplayWidget(0.9)),
                Center(child: ChineseMoneyDisplayWidget(0.17)),
                Center(child: ChineseMoneyDisplayWidget(0.1796)),
                Center(child: ChineseMoneyDisplayWidget(6)),
                Center(child: ChineseMoneyDisplayWidget(6.1)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    9.16,
                    decimalFontsize: 29,
                    integerColor: Color.fromARGB(255, 0, 150, 187),
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(9.1671)),
                Center(child: ChineseMoneyDisplayWidget(10)),
                Center(child: ChineseMoneyDisplayWidget(10.7)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    10.71,
                    integerFontsize: 25,
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(999)),
                Center(child: ChineseMoneyDisplayWidget(999.7)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    999.99,
                    decimalDigitColor: Color.fromARGB(255, 80, 0, 133),
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(1000)),
                Center(child: ChineseMoneyDisplayWidget(1995.0)),
                Center(child: ChineseMoneyDisplayWidget(1995.07)),
                Center(child: ChineseMoneyDisplayWidget(6666.66)),
                Center(child: ChineseMoneyDisplayWidget(9999)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    99999,
                    currencyFontWeight: FontWeight.w900,
                    currencySymbolColor: Color.fromARGB(255, 22, 0, 148),
                    currencySymbolFontsize: 56,
                  ),
                ),
                Center(child: ChineseMoneyDisplayWidget(999999)),
                Center(child: ChineseMoneyDisplayWidget(996786)),
                Center(
                  child: ChineseMoneyDisplayWidget(
                      9999999), // maxNum defaults to 10000.
                ),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    9999999,
                    maxNum:
                        10000001, // Set a custom maximum number for display.
                  ),
                ),
                Center(
                  child: ChineseMoneyDisplayWidget(
                    9999999.97,
                    maxNum: 10000001,
                    integerColor: Colors.blue,
                    integerFontsize: 25,
                    decimalFontsize: 21,
                    currencySymbolColor: Colors.pink,
                    tenThousandSymbol: 'w',
                    tenThousandSymbolColor: Color.fromARGB(255, 0, 156, 78),
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
